README.yaml

```diff
@@ -0,0 +1,247 @@
+
+  The module accepts the following:
+
+  - Paths to local and remote YAML configuration template files for [Terraform](https://www.terraform.io/) and [helmfile](https://github.com/roboll/helmfile) stacks
+  - Map of parameters for interpolation within the YAML config templates (optional)
+  - Component type (`terraform` or `helmfile`) for which to return configurations
+  - Component name for which to return the configurations (deep merged variables and Terraform backend attributes)
+
+  The module supports a top-level `import` attribute in  YAML configuration templates, which will include the files and perform deep merging.
+  Up to 10 levels of imports are supported, and all imported maps are deep merged into a final stack configuration map.
+
+  For example, if you have a config file like this (e.g. `myconfig.yaml`):
+
+    ```yaml
+      import:
+        - file1
+        - file2
+    ```
+
+  Then, the module will deep merge `file1.yaml` and `file2.yaml` into `myconfig.yaml`.
+
+  __Note:__ Do not include the extensions (e.g. `.yaml`) in the imports.
+
+  The module performs two-phase deep merging:
+
+  - All maps from the provided and all imported YAML config files are deep merged first into the final stack configuration
+  - The `vars` and `backend` sections from different parts of YAML configs are then deep merged into the final variables and Terraform backend configurations
+    for the specified component type and component name
+
+  The module returns the following:
+
+  - List of all imported files
+  - The final deep merged stack configuration from all provided and imported YAML files
+  - The final configuration of all variables (`vars`) for the specified component type and component name
+  - The final configuration of all Terraform backend attributes (`backend`) for the specified component
+
+  ### Attributions
+
+  Big thanks to [Imperative Systems Inc.](https://github.com/Imperative-Systems-Inc)
+  for the excellent [deepmerge](https://github.com/Imperative-Systems-Inc/terraform-modules/tree/master/deepmerge) Terraform module
+  to perform a deep map merge of standard Terraform maps and objects.
+
+# How to use this module. Should be an easy example to copy and paste.
+usage: |-
+  For a complete example, see [examples/complete](examples/complete).
+
+  For automated tests of the complete example using [bats](https://github.com/bats-core/bats-core) and [Terratest](https://github.com/gruntwork-io/terratest)
+  (which tests and deploys the example on Datadog), see [test](test).
+
+
+# Example usage
+examples: |-
+
+  ```hcl
+  module "yaml_stack_config" {
+    source = "cloudposse/stack-config/yaml"
+    # Cloud Posse recommends pinning every module to a specific version
+    # version     = "x.x.x"
+
+    map_config_local_base_path = "./config"
+
+    map_config_paths = [
+      "my-stack.yaml"
+    ]
+
+    component_type = "terraform"
+    component      = "my-vpc"
+
+    context = module.this.context
+  }
+  ```
+
```

Comment:
We also need to explain the structure of a stack configuration. 

We should also consider adding a `version` to the stack YAML configuration.